pkg/tasks/component.go

```diff
@@ -0,0 +1,106 @@
+package tasks
+
+import (
+	"context"
+	"database/sql"
+
+	cs "github.com/cloudtrust/common-service"
+	"github.com/cloudtrust/common-service/database"
+	"github.com/cloudtrust/common-service/log"
+	api "github.com/cloudtrust/keycloak-bridge/api/tasks"
+	"github.com/cloudtrust/keycloak-bridge/internal/keycloakb"
+	"github.com/cloudtrust/keycloak-client"
+)
+
+// Component interface exposes methods used by the bridge API
+type Component interface {
+	GetActions(ctx context.Context) ([]api.ActionRepresentation, error)
+	DeleteDeniedTermsOfUseUsers(ctx context.Context) error
+}
+
+// KeycloakClient interface
+type KeycloakClient interface {
+	DeleteUser(accessToken string, realmName, userID string) error
+	GetExpiredTermsOfUseAcceptance(accessToken string) ([]keycloak.DeletableUserRepresentation, error)
+}
+
+// UsersDetailsDBModule is the interface from the users module
+type UsersDetailsDBModule interface {
+	DeleteUserDetails(ctx context.Context, realm string, userID string) error
+}
+
+type component struct {
+	keycloakClient KeycloakClient
+	usersDBModule  UsersDetailsDBModule
+	eventDBModule  database.EventsDBModule
+	logger         log.Logger
+}
+
+// NewComponent returns a component
+func NewComponent(keycloakClient KeycloakClient, usersDBModule UsersDetailsDBModule, eventDBModule database.EventsDBModule,
+	logger log.Logger) Component {
+	return &component{
+		keycloakClient: keycloakClient,
+		usersDBModule:  usersDBModule,
+		eventDBModule:  eventDBModule,
+
+		logger: logger,
+	}
+}
+
+func (c *component) reportEvent(ctx context.Context, apiCall string, values ...string) {
+	errEvent := c.eventDBModule.ReportEvent(ctx, apiCall, "scheduled-tasks", values...)
+	if errEvent != nil {
+		//store in the logs also the event that failed to be stored in the DB
+		keycloakb.LogUnrecordedEvent(ctx, c.logger, apiCall, errEvent.Error(), values...)
+	}
+}
+
+func (c *component) GetActions(ctx context.Context) ([]api.ActionRepresentation, error) {
```

Comment:
Do we have sync with Fratt so we can really set those rights ?

Reply:
I removed GetActions in both tasks and communications...
I also added the only action from tasks in the GetActions from management, as we did for actions from communications